scapy/layers/inet.py

```diff
@@ -521,11 +521,28 @@ def hashret(self):
     def answers(self, other):
         if not isinstance(other, TCP):
             return 0
+        # RST packets don't get answers
+        if other.flags.R:
+            return 0
+        if self.flags.S:
+            # SYN packets without ACK are not answers
```

Comment:
Thanks for the heads up!

From what I've read, TCP four-way handshakes could happen in two different ways:
```
    (1)            (2)
A         B    A         B
|-->-S->--|    |-->-S->--|    (p1)
|--<-A-<--|    |--<-S-<--|    (p2)
|--<-S-<--|    |->-S/A->-|    (p3)
|-->-A->--|    |-<-S/A-<-|    (p4)
```

In the first case, `p2` answers `p1` and `p4` answers `p3`. One could argue, however, that `p3` also answers `p1`, which should have an impact on `sr` and `srp` used with `multi=True`. While that is true, it would be difficult to handle in Scapy since `p1` does not contain a reference to `p3` sequence number and  `p3` does not contain a reference to `p1` sequence number, so we would only be able to check the IP addresses and TCP ports, and maybe we would have to use the timestamps to make sure we're not seeing an (IP, port, IP, port) reuse: I don't think it's worth it.

The second case would only happen when both A and B open at the same time a connection to each other (RFC 793, section 3.4, "Simultaneous Connection Synchronization"). In that case, I would say that `p2` does not answer `p1`: they just happen to be simultaneously sent. `p3` answers `p1` and `p4` answers `p2`, and that should just work with the current code.

What do you think?